recipes/petsc/build.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+export PETSC_DIR=$SRC_DIR
+export PETSC_ARCH=arch-conda-c-opt
+
+if [[ $(uname) == Darwin ]]; then
+    SO=dylib
+else
+    SO=so
+fi
+
+$PYTHON ./configure \
+  --with-fc=0 \
+  --with-debugging=0 \
+  --COPTFLAGS=-O3 \
+  --CXXOPTFLAGS=-O3 \
+  --LIBS=-Wl,-rpath,$PREFIX/lib \
+  --with-blas-lapack-lib=libopenblas.$SO \
+  --with-cmake=0 \
+  --with-hwloc=0 \
+  --with-ssl=0 \
+  --with-x=0 \
+  --prefix=$PREFIX
+
+sedinplace() { [[ $(uname) == Darwin ]] && sed -i "" $@ || sed -i"" $@; }
+for path in $PETSC_DIR $PREFIX; do
+    sedinplace s%$path%\${PETSC_DIR}%g $PETSC_ARCH/include/petsc*.h
+done
+
+make
+make install
+
+rm -fr $PREFIX/bin
```

Comment:
Do we really want to delete the entire `bin` and `share` directories? Or are there just particular files that PETSc introduces?